staging/src/k8s.io/apiserver/pkg/server/filters/timeout.go

```diff
@@ -91,18 +92,30 @@ func (t *timeoutHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	result := make(chan interface{})
+	type result struct {
+		err   interface{}
+		stack string
+	}
+	resCh := make(chan result)
 	tw := newTimeoutWriter(w)
 	go func() {
 		defer func() {
-			result <- recover()
+			res := result{
+				err: recover(),
+			}
+			if res.err != nil {
+				buf := make([]byte, 2048)
+				n := runtime.Stack(buf, false)
+				res.stack = string(buf[:n])
+			}
+			resCh <- res
 		}()
 		t.handler.ServeHTTP(tw, r)
 	}()
 	select {
-	case err := <-result:
-		if err != nil {
-			panic(err)
+	case res := <-resCh:
+		if res.err != nil {
+			panic(fmt.Sprintf("%v: %s", res.err, res.stack))
```

Comment:
put the stack on its own line: `fmt.Sprintf("%v\n%s", res.err, res.stack)` so that if things higher in the chain do the same thing with the propagated panic, it composes correctly